### ISSUE TITLE
issue with Java 1.8 and QueryDSL

### ISSUE THREAD

**Issue by pablo-parra** [oasp/oasp4j#599]
When trying to use java 1.8 in an Oasp4j project, and using CobiGen to generate the component code, in POST requests that require the usage of queryDsl library, we are facing the error:

```
java.lang.IllegalArgumentException: null
```

The error is thrown in the _DaoImpl_ when using the `com.mysema.query.alias.Alias` in the line 

```
DemoEntity demo = Alias.alias(DemoEntity.class);
```

When debugging the error, internally it is thrown in the _guava-17.0 library_, _com.google.common.cache.LocalCache.class_, line 2283 → `statsCounter.recordMisses(1)`

Going back to the Oasp4j app seems also that the problem is related to the Generic interfaces like `net.sf.mmm.util.entity.api.MutableGenericEntity.class`, `net.sf.mmm.util.entity.api.GenericEntity.class`,  `net.sf.mmm.util.lang.api.attribute.AttributeReadId.class` and so forth. In those classes a generic `ID` type is managed and seems that, for some reason, is not correctly handled in Java 1.8

With the current interfaces definition the error appears:
```
public interface MutableGenericEntity<ID> extends GenericEntity<ID> {
  void setId(ID id);
...
}
```

I have tried the following, and the error disappears:

```
public interface MyMutableGenericEntity<ID> extends MyGenericEntity<ID> {
  void setId(Long id);
...
}
```

I’m not sure why this is happening only with Java 1.8

**Comment by vapadwal** [comment]
Hi Pablo,

As this issue is been fixed, can you please close it.

Thanks
Vaibhav

**Comment by sjimenez77** [comment]
Hi @vapadwal, 

This issue finally was not solved in OASP4J, only in My Thai Star. 
It has just happened again in a new project here in Valencia. 

Could you fix this and make a PR?

Thanks!

**Comment by vapadwal** [comment]
This issue is producible only when we use oasp4j older template (tested it with - 2.3.0 ) . This issue is not been produced with latest oasp4j template. So current template 2.5.0 doesn't have this issue.

**Comment by pablo-parra** [comment]
I have just reproduced the error starting from an empty app. As indicated in the issue fix (https://github.com/oasp/my-thai-star/pull/31/files#diff-88df895563dd032c84e3d0a66209313c) the problem was related to that cglib version. It is supposed to be `<version>3.2.4</version>` but with the 2.5.0 archetype there is no version change in the cglib library. In addition we need to change the java version to 1.8. Then, with a simple Entity and using Cobigen I have created all the services. The DaoImpl will use querydsl to filter the search calls using a SearchCriteria. Doing this and with a POST call (where querydsl has to participate in the filtering) we get the following error: 
`Caused by: java.lang.IllegalArgumentException: null
	at org.objectweb.asm.ClassReader.<init>(Unknown Source) ~[asm-4.2.jar:4.2]
	at org.objectweb.asm.ClassReader.<init>(Unknown Source) ~[asm-4.2.jar:4.2]
	at org.objectweb.asm.ClassReader.<init>(Unknown Source) ~[asm-4.2.jar:4.2]
	at net.sf.cglib.proxy.BridgeMethodResolver.resolveAll(BridgeMethodResolver.java:61) ~[cglib-3.1.jar:na]
	at net.sf.cglib.proxy.Enhancer.emitMethods(Enhancer.java:911) ~[cglib-3.1.jar:na]
	at net.sf.cglib.proxy.Enhancer.generateClass(Enhancer.java:498) ~[cglib-3.1.jar:na]
	at net.sf.cglib.core.DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25) ~[cglib-3.1.jar:na]
	at net.sf.cglib.core.AbstractClassGenerator.create(AbstractClassGenerator.java:216) ~[cglib-3.1.jar:na]
	at net.sf.cglib.proxy.Enhancer.createHelper(Enhancer.java:377) ~[cglib-3.1.jar:na]
	at net.sf.cglib.proxy.Enhancer.create(Enhancer.java:285) ~[cglib-3.1.jar:na]
	at com.mysema.query.alias.AliasFactory.createProxy(AliasFactory.java:140) ~[querydsl-core-3.7.1.jar:na]
	at com.mysema.query.alias.AliasFactory$1.load(AliasFactory.java:53) ~[querydsl-core-3.7.1.jar:na]
	at com.mysema.query.alias.AliasFactory$1.load(AliasFactory.java:50) ~[querydsl-core-3.7.1.jar:na]
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3524) ~[guava-17.0.jar:na]
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2317) ~[guava-17.0.jar:na]
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2280) ~[guava-17.0.jar:na]
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2195) ~[guava-17.0.jar:na]
	... 109 common frames omitted`

So the issue is still there.

Simply adding the `<version>3.2.4</version>` to the cglib dependency the issue is solved, but as I have mentioned the 2.5.0 archetype does not add this version to the core/pom.xml by default (or to the parent)
